packages/gafl-webapp-service/src/handlers/authentication-handler.js

```diff
@@ -20,18 +21,50 @@ export default async (request, h) => {
     .createBirthDateValidator(Joi)
     .validateAsync(`${payload['date-of-birth-year']}-${payload['date-of-birth-month']}-${payload['date-of-birth-day']}`)
   const postcode = await validation.contact.createUKPostcodeValidator(Joi).validateAsync(payload.postcode)
-  const { referenceNumber } = await request.cache().helpers.status.getCurrentPermission()
+
+  const permission = await request.cache().helpers.status.getCurrentPermission()
+  const referenceNumber = payload.referenceNumber || permission.referenceNumber
 
   // Authenticate
   const authenticationResult = await salesApi.authenticate(referenceNumber, dateOfBirth, postcode)
 
+  const linkInactive = async reason => {
+    await request.cache().helpers.status.setCurrentPermission({
+      referenceNumber,
+      authentication: {
+        authorized: false,
+        reason: reason,
+        endDate: authenticationResult.permission.endDate
+      }
+    })
+    return h.redirect(RENEWAL_INACTIVE.uri)
+  }
+
   if (!authenticationResult) {
-    await request.cache().helpers.status.setCurrentPermission({ authentication: { authorized: false } })
-    return h.redirect(IDENTIFY.uri.replace('{referenceNumber}', referenceNumber))
+    payload.referenceNumber = referenceNumber
+    await request.cache().helpers.page.setCurrentPermission(IDENTIFY.page, { payload, error: { referenceNumber: 'string.invalid' } })
+    await request.cache().helpers.status.setCurrentPermission({ referenceNumber, authentication: { authorized: false } })
+    return h.redirect(IDENTIFY.uri)
   } else {
-    await setUpCacheFromAuthenticationResult(request, authenticationResult)
-    await setUpPayloads(request)
-    await request.cache().helpers.status.setCurrentPermission({ authentication: { authorized: true } })
-    return h.redirect(CONTROLLER.uri)
+    // Test for 12 month licence
+    if (
+      authenticationResult.permission.permit.durationDesignator.description === 'M' &&
+      authenticationResult.permission.permit.durationMagnitude === 12
```

Comment:
I think its ok as a single use - I searched the code base and this is the only use of this logic as f(permit). Either you are simply going to copy the expression into a single use function or you use an intermediate variable throughout permit = authenticationResult.permission.permit which I don't much like either.

Reply:
I do agree with Phil here.  Not insisting we do this now but I can see that this will end up being refactored into the business-rules-lib - for example, determining whether a licence is eligible to allow for a DD to be setup will also use this logic.  

This is somewhat fragile as it doesn't account for ISO durations which might use different UOM (e.g. 1Y).   In the Sales API I've used moment to construct a duration based on the ISO standard:
```
moment.duration(`P${permit.durationMagnitude}${permit.durationDesignator.description}`)
```

Reply:
Yes - we never attempted to make the front end completely data driven. That is a different project. It has very specific rules and user interactions with a 12 month licence as compared to an 8 day licence for instance. At some point you have to stop generalising and deliver. Can we move on please.